internal/services/v1/errors.go

```diff
@@ -177,6 +177,62 @@ func (err ErrDuplicateRelationshipError) GRPCStatus() *status.Status {
 	)
 }
 
+// ErrCouldNotTransactionallyDelete indicates that a deletion could not occur transactionally.
+type ErrCouldNotTransactionallyDelete struct {
+	error
+	limit  uint32
+	filter *v1.RelationshipFilter
+}
+
+// NewCouldNotTransactionallyDeleteErr constructs a new could not transactionally deleter error.
+func NewCouldNotTransactionallyDeleteErr(filter *v1.RelationshipFilter, limit uint32) ErrCouldNotTransactionallyDelete {
+	return ErrCouldNotTransactionallyDelete{
+		error: fmt.Errorf(
+			"found more than %d relationships to be deleted and partial deletion was not requested",
+			limit,
+		),
+		limit:  limit,
+		filter: filter,
+	}
+}
+
+// GRPCStatus implements retrieving the gRPC status for the error.
+func (err ErrCouldNotTransactionallyDelete) GRPCStatus() *status.Status {
```

Comment:
I almost feel like asking for a unit test. It's pretty simple logic, but regressions here means API breakage

Reply:
We have a unit test for this: this error is checked in one of the service tests